blockchain_connector/avalon_blockchain_connector/fabric/fabric_connector.py

```diff
@@ -75,17 +76,43 @@ def sync_worker(self):
         """
         Check for existing worker and update worker to fabric blockchain
         """
-        worker_type = WorkerType.TEE_SGX
+        # First get all the workers from the blockchain
         req_id = 15
-        lookup_result = self.__jrpc_worker.worker_lookup(
+        bc_workers = self.__fabric_worker.worker_lookup(
+            worker_type=WorkerType.TEE_SGX,
+        )
+        logging.info("Workers in blockchain {}".format(
+            bc_workers
+        ))
+        # Set worker status to Decommissioned.
+        if bc_workers and bc_workers[0] > 0:
+            for worker in bc_workers[2]:
+                update_status = self.__fabric_worker.worker_set_status(
+                    worker,
+                    WorkerStatus.DECOMMISSIONED
```

Comment:
Updating state in a block chain is costly operation and will take time. So one option is to selectively update the state in block chain (to decommissioned) only if worker is not present in shared KV. We need to make the logic of syncing workers between shared KV and block chain uniform. So this could be handled as part of it